gen/container_test.go

```diff
@@ -369,3 +369,465 @@ func TestListOfStructs(t *testing.T) {
 		assertRoundTrip(t, &tt.s, tt.v, "Graph")
 	}
 }
+
+func TestCrazyTown(t *testing.T) {
+	tests := []struct {
+		desc string
+		x    tc.ContainersOfContainers
+		v    wire.Value
+	}{
+		{
+			"ListOfLists",
+			tc.ContainersOfContainers{
+				ListOfLists: [][]int32{
+					{1, 2, 3},
+					{4, 5, 6},
+				},
+			},
+			wire.NewValueStruct(wire.Struct{Fields: []wire.Field{
+				{ID: 1, Value: wire.NewValueList(wire.List{
+					ValueType: wire.TList,
+					Size:      2,
+					Items: wire.ValueListFromSlice([]wire.Value{
+						wire.NewValueList(wire.List{
+							ValueType: wire.TI32,
+							Size:      3,
+							Items: wire.ValueListFromSlice([]wire.Value{
+								wire.NewValueI32(1),
+								wire.NewValueI32(2),
+								wire.NewValueI32(3),
+							}),
+						}),
+						wire.NewValueList(wire.List{
+							ValueType: wire.TI32,
+							Size:      3,
+							Items: wire.ValueListFromSlice([]wire.Value{
+								wire.NewValueI32(4),
+								wire.NewValueI32(5),
+								wire.NewValueI32(6),
+							}),
+						}),
+					}),
+				})},
+			}}),
+		},
+		{
+			"ListOfSets",
+			tc.ContainersOfContainers{
+				ListOfSets: []map[int32]struct{}{
+					map[int32]struct{}{
```

Comment:
you shouldn't need this `map[int32]struct{}` here,
https://play.golang.org/p/q-rkKLbw6z